crates/uv-build-frontend/src/lib.rs

```diff
@@ -570,7 +580,13 @@ impl SourceBuild {
                             );
                         }
                     }
-                    default_backend.clone()
+                    let mut backend = default_backend.clone();
+                    // Apply extra_build_dependencies
+                    // TODO(Gankra): should Sources/Indexes be applied on this path?
+                    backend
+                        .requirements
+                        .extend(extra_build_dependencies.into_iter().map(Requirement::from));
+                    backend
                 };
```

Comment:
Sad path #1 where I probably need to handle sources where previously they were always irrelevant (pyproject.toml but missing `[build-system]`)